google/data_source_google_service_account.go

```diff
@@ -44,15 +44,15 @@ func dataSourceGoogleServiceAccountRead(d *schema.ResourceData, meta interface{}
 	config := meta.(*Config)
 
 	// Get the project from the resource or fallback to the project
-	// in the provider configuration
-	project, err := getProject(d, config)
+	// in the provider configuration. If neither exist use an empty project
+	// when building the service account FQN
+	project, _ := getProject(d, config)
```

Comment:
Generally I'm not too into the idea of ignoring errors. What if `serviceAccountFQN` took `d` and `config` as parameters, and then it could do the getProject call at the time where there does actually need to be a project set?

Reply:
That seems much better, I'll do that!

Reply:
@danawillow I have made the requested changes, and it looks much cleaner now, thanks :)